amundsen_common/utils/streams.py

```diff
@@ -0,0 +1,388 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import threading
+from typing import (Any, AsyncIterator, Callable, Collection, Iterable,
+                    Iterator, List, Optional, Tuple, TypeVar, Union)
+
+from typing_extensions import Final, final
+
+LOGGER = logging.getLogger(__name__)
+
+
+V = TypeVar('V')
+R = TypeVar('R')
+
+
+def one(ignored: Any) -> int:
+    return 1
+
+
+class PeekingIterator(Iterator[V]):
```

Comment:
so I don't have much context for all the code. If it is for gremlin lib, could we move it to https://github.com/amundsen-io/amundsengremlin ? The common lib is only used to store common model for all the repos. Let me know what you thinks.

Reply:
Yeah...I was wondering myself if this is useful in anyone else's databuilder. We use this in the gremlin code, but also for some of our databuilder jobs to chunk the results (e.g. 1000 or 10,000. Unlimited size batches can be problematic in some circumstances). Do y'all ever use airflow to explicitly set batch sizes?

Reply:
We define the batch / transaction size through config during neo4j ingestion (https://github.com/amundsen-io/amundsendatabuilder/blob/master/databuilder/publisher/neo4j_csv_publisher.py#L95). Otherwise the neo4j will hang without batch size setting. So given we haven't had a clear use case, I would suggest moving the code to gremlin repo? we could always use gremlin as an extra library deps in databuilder if we plan to enhance and build neptune publisher there.  cc @AndrewCiambrone

Reply:
Sure, sounds good to me! Thanks for the colour, Tao!

- [ ] streams to gremlin repo